src/classes/CON_PrimaryAffls_TEST.cls

```diff
@@ -768,51 +768,492 @@ private with sharing class CON_PrimaryAffls_TEST {
+
+        List<Affl_Mappings__c> mappings = new List<Affl_Mappings__c>();
+        mappings.add(new Affl_Mappings__c(Name = 'Business Organization', Account_Record_Type__c = 'Business_Organization', Primary_Affl_Field__c = prefix + 'Gender__c'));
+        mappings.add(new Affl_Mappings__c(Name = 'Household Account', Account_Record_Type__c = 'HH_Account', Primary_Affl_Field__c =  prefix + 'Primary_Household__c'));
+        insert mappings;
+
+        // insert business account
+        Account businessAccount1 = UTIL_UnitTestData_TEST.getMultipleTestAccounts(1, UTIL_Describe_API.getBizAccRecTypeID())[0];
+        insert businessAccount1;
+
+        for (Contact con : testContactsList){
+            con.Primary_Organization__c = businessAccount1.Id;
+            con.Gender__c = 'Decline to State';
+        }
+
+        Test.startTest();
+        try {
+            update testContactsList;
+        } catch (ERR_ExceptionHandler.AffAccountRecordTypeMappingsException err){
+            System.assertEquals(CON_PrimaryAffls_TEST.getExpectedInvalidFieldMappingErrorMessage(mappings[0]), err.getMessage(), 'Invalid Record Type mapping should throw an error when validation is enabled.');
+        }
+        
+        Test.stopTest();
+    }
+
+    /*************************************************************************************************************
+    * @description Test case to verify that updating a primary affiliation field on Contact with record type
+    * validation enabled, a valid Contact field mapping to a non-lookup field and invalid Record Type mapping throws 
+    * an error when the mapped field is populated.
+    **************************************************************************************************************/
+    @isTest 
+    private static void updateContactAfflInvalidRTAndInvalidFieldMappingFieldPopulatedInvalidRecordTypeError(){
+        List<Contact> testContactsList = UTIL_UnitTestData_TEST.getMultipleTestContacts(100);
+        insert testContactsList;
+
+        UTIL_CustomSettings_API.getSettingsForTests(new Hierarchy_Settings__c(Affiliation_Record_Type_Enforced__c = true));
+
+        String namespace = UTIL_Namespace.getNamespace();
+        String prefix = (String.isNotBlank(namespace)) ? namespace + '__' : '';
+
+        List<Affl_Mappings__c> mappings = new List<Affl_Mappings__c>();
+        mappings.add(new Affl_Mappings__c(Name = 'Business Organization', Account_Record_Type__c = 'Not a Valid RecordType', Primary_Affl_Field__c = prefix + 'Gender__c'));
+        mappings.add(new Affl_Mappings__c(Name = 'Household Account', Account_Record_Type__c = 'HH_Account', Primary_Affl_Field__c =  prefix + 'Primary_Household__c'));
+        insert mappings;
+
+        for (Contact con : testContactsList){
+            con.Gender__c = 'Decline to State';
+        }
+
+        Test.startTest();
+        try {
+            update testContactsList;
```

Comment:
Another consideration of the single-line test block and the `System.assert(false,String)`